tests/api/v2/handlers/test_agents_api.py

```diff
@@ -0,0 +1,256 @@
+
+
+@pytest.fixture
+def new_agent_payload():
+    return {
+        'paw': '456',
+        'sleep_min': 3,
+        'sleep_max': 6,
+        'watchdog': 0,
+        'group': 'red',
+        'architecture': '',
+        'platform': 'linux',
+        'server': 'http://localhost:8000',
+        'upstream_dest': 'http://localhost:8000',
+        'username': 'test',
+        'location': './splunkd',
+        'pid': 1,
+        'ppid': 1,
+        'trusted': True,
+        'executors': ['sh'],
+        'privilege': 'User',
+        'exe_name': 'splunkd',
+        'host': 'test_agent',
+        'contact': 'HTTP',
+        'proxy_receivers': {},
+        'proxy_chain': [],
+        'origin_link_id': 0,
+        'deadman_enabled': True,
+        'available_contacts': ['HTTP'],
+        'host_ip_addrs': []
+    }
+
+
+@pytest.fixture
+def expected_new_agent_dump(new_agent_payload, mocker, mock_time):
+    with mocker.patch('datetime.datetime') as mock_datetime:
+        mock_datetime.return_value = mock_datetime
+        mock_datetime.now.return_value = mock_time
+        agent = Agent().load(new_agent_payload)
+        return agent.schema.dump(agent)
+
+
+@pytest.fixture
+def test_agent(loop, mocker, mock_time):
+    with mocker.patch('datetime.datetime') as mock_datetime:
+        mock_datetime.return_value = mock_datetime
+        mock_datetime.now.return_value = mock_time
+        test_agent = Agent(paw='123', sleep_min=2, sleep_max=8, watchdog=0, executors=['sh'], platform='linux')
+        loop.run_until_complete(BaseService.get_service('data_svc').store(test_agent))
+        return test_agent
+
+
+@pytest.fixture
+def test_executor(test_agent):
+    return ExecutorSchema().load(dict(timeout=60, platform=test_agent.platform, name='linux',
+                                      command='ls'))
+
+
+@pytest.fixture
+def deploy_ability(test_executor, loop):
+    ability = AbilitySchema().load(dict(ability_id='123',
+                                        tactic='persistence',
+                                        technique_id='auto-generated',
+                                        technique_name='auto-generated',
+                                        name='test deploy command',
+                                        description='test ability',
+                                        executors=[ExecutorSchema().dump(test_executor)]))
+    loop.run_until_complete(BaseService.get_service('data_svc').store(ability))
+    return ability
+
+
+@pytest.fixture
+def raw_ability(deploy_ability, test_executor):
+    raw_ability = {
+        'name': deploy_ability.name,
+        'platform': test_executor.platform,
+        'executor': test_executor.name,
+        'description': deploy_ability.description,
+        'command': test_executor.command,
+        'variations': [{'description': v.description, 'command': v.raw_command} for v in test_executor.variations]
+    }
+    return raw_ability
+
+
+@pytest.fixture
+def app_config():
```

Comment:
If memory serves me correctly, some of these configs were used in other API v2 pytests. Is there a way you can move them to a common file like conftest or something similar? That way we're not duplicating fixtures unnecessarily

Reply:
I did some searching, but I don't think any other test is currently using these fixtures. I believe that I still need to write tests for config_api.py though, so I can do some refactoring in that PR if there's any overlap

Reply:
Whoops, I overlooked test/api/v2/managers. These configs have been moved to conftest.py!